eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/list/smart/AbstractClickableSmartListItem.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.list.smart;
+
+import android.support.annotation.LayoutRes;
+import android.view.View;
+
+import com.schedjoules.eventdiscovery.framework.list.ListItem;
+import com.schedjoules.eventdiscovery.framework.utils.smartview.ClickableSmartView;
+import com.schedjoules.eventdiscovery.framework.utils.smartview.SmartClickable;
+import com.schedjoules.eventdiscovery.framework.utils.smartview.SmartViewOnClickAction;
+
+
+/**
+ * Abstract class for {@link ListItem}s that work with {@link ClickableSmartView}s.
+ * <p>
+ * Note: Usually list items' 'equality' should be based on the data/model objects (type D) equality, so make sure that the data type equality is defined.
+ *
+ * @author Gabor Keszthelyi
+ */
+public abstract class AbstractClickableSmartListItem<D, V extends View & ClickableSmartView<D>> implements ListItem<V>, SmartClickable<D>
+{
+    private final D mItemData;
+    private final int mLayout;
+
+    // Note: In some circumstances it is useful if the action can be set after the item has been created, that's why it's not final.
```

Comment:
This has been removed, so no such usage currently, but if the `ListItem`s are prepared on background thread, then we may need to set the listeners after they have been created. That is the case in the Event list currently, but those are not smart yet.